venues/ICML.cc/2018/Workshop/NAMPI/python/config.py

```diff
@@ -0,0 +1,101 @@
+'''
+NAMPI - an ICML workshop
+
+2018-05-08: setup workshop recruiting infrastructure
+
+
+'''
+
+import openreview
+from openreview import invitations
+import os
+
+CONFERENCE_ID = 'ICML.cc/2018/Workshop/NAMPI'
+PROGRAM_CHAIRS_ID = CONFERENCE_ID + '/Program_Chairs'
+REVIEWERS_ID = CONFERENCE_ID + '/Reviewers'
+REVIEWERS_INVITED_ID = REVIEWERS_ID + '/Invited'
+REVIEWERS_DECLINED_ID = REVIEWERS_ID + '/Declined'
+
+
+'''
+Group definitions.
+'''
+CONFERENCE = openreview.Group(**{
```

Comment:
we don't need uppercase variable names

Reply:
I don't have strong opinions about this, but there is at least some argument for having uppercase variable names in this context:

https://www.python.org/dev/peps/pep-0008/#constants

it talks about using uppercase variable names "at the module level". It's not a perfect comparison, but I do tend to think of the /python directories as being a sort of module for the conference.

If you disagree I don't mind changing it. just wanted to share my reasons

Reply:
Well I see as constants, variables that won't change like the strings to define the group ids or invitations ids. 

These are instances of an object that will change in the future. And I don't think the conference group is a constant. May be we can define here the base config of the conference group but not the conference group itself, because this group will change in the future or it is already created.

Reply:
I'm not sure if we need to save the group config as a constant. May be we just need to create the group in the setup script and that's all.

Reply:
Good point - I'll change this to reflect the group parameters instead of
the object itself.

Reply:
oops, our responses got out of sync.

Let me think about this a little more. maybe you're right

Reply:
Now that I've gathered my thoughts:

The reason I did it this way is to avoid creating a setup script altogether. Setup scripts are run once and never run again, and they rarely stay up to date. They're also unwieldy: a lot of the time we want to make small changes in the setup but can't run the setup script because it would break other parts of the conference.

by putting object definitions (groups, invitations, etc) in the config file, users can import the file and then manipulate the contents as they see fit. Yes, there is a risk that the objects in the config file won't be up-to-date, but there is the same risk with a setup script.

in my opinion, the solution to this is a cultural one, not a technical one: we have to make sure that admins / workflow engineers understand that the config file represents the state that the conference is supposed to be in, and that if they make changes, they need to commit them to the repo. Of course they could make changes outside of this script, but at least this way there is a single location where the parameters are *supposed* to be stored.

Reply:
We can talk about this afternoon if you want and discuss the new changes that you want to do.

Reply:
Ok, that sounds good. I'll send you an email to figure out a good time.